### PR TITLE
Update help and error messages of cli

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Command line interface: Update help and error messages.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/cli/arguments/run.py [hunk 1]
```diff
@@ -51,14 +51,11 @@ def add_server_arguments(parser: argparse.ArgumentParser):
...
-        help="Start the web server api in addition to the input channel.",
+        help="Start the web server API in addition to the input channel.",
     )
     server_arguments.add_argument(
         "--remote-storage",
-        help="Set the remote location where models are stored. "
-        "E.g. on AWS. If nothing is configured, the "
-        "server will only serve the models that are "
-        "on disk in the configured model path.",
+        help="Set the remote location where your Rasa model is stored, " "e.g. on AWS.",
```
Comment (erohmensing) [inline rasa/cli/arguments/run.py]:
```suggestion
        help="Set the remote location where your Rasa model is stored, e.g. on AWS.",
```

File: rasa/cli/arguments/test.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def add_test_core_argument_group(
     parser.add_argument(
         "--url",
         type=str,
-        help="If supplied, downloads a story file from a URL and "
+        help="If supplied, downloads a story file from an URL and "
```
Comment (erohmensing) [inline rasa/cli/arguments/test.py]:
```suggestion
        help="If supplied, downloads a story file from a URL and "
```
English is weird 😆

File: rasa/cli/data.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def add_subparser(
         "nlu",
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
         parents=parents,
-        help="Convert NLU training data between markdown and json",
+        help="Converts NLU data between markdown and json.",
```
Comment (erohmensing) [inline rasa/cli/data.py]:
```suggestion
        help="Converts NLU data between Markdown and json formats.",
```

File: rasa/cli/data.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def add_subparser(
         "split",
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
         parents=parents,
-        help="Split Rasa data in training and test data",
+        help="Splits Rasa data in training and test data.",
```
Comment (erohmensing) [inline rasa/cli/data.py]:
```suggestion
        help="Splits Rasa data into training and test data.",
```

File: rasa/cli/data.py [hunk 5]
```diff
@@ -55,8 +55,8 @@ def add_subparser(
         "nlu",
         parents=parents,
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        help="Perform a split of your NLU data according to the specified "
-        "percentages",
+        help="Performs a split of your NLU data according to the specified "
```
Comment (erohmensing) [inline rasa/cli/data.py]:
```suggestion
        help="Performs a split of your NLU data into training and test data "
```

File: rasa/cli/data.py [hunk 6]
```diff
@@ -55,8 +55,8 @@ def add_subparser(
...
+        "percentages.",
```
Comment (erohmensing) [inline rasa/cli/data.py]:
```suggestion
        "according to the specified percentages.",
```

File: rasa/cli/interactive.py [hunk 7]
```diff
@@ -32,7 +32,7 @@ def add_subparser(
         conflict_handler="resolve",
         parents=parents,
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        help="Train a Rasa Core model with interactive learning",
+        help="Trains a Rasa Core model in an interactive learning session.",
```
Comment (erohmensing) [inline rasa/cli/interactive.py]: Does this train the model or just create new training data as above?
Also, maybe something that says ```Uses the RegexInterpreter, i.e. `/intent` input format``` or something like that

File: rasa/cli/interactive.py [hunk 8]
```diff
@@ -22,7 +22,7 @@ def add_subparser(
         conflict_handler="resolve",
         parents=parents,
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        help="Teach the bot with interactive learning",
+        help="Starts an interactive learning session to create new training data by chatting.",
```
Comment (erohmensing) [inline rasa/cli/interactive.py]: maybe "new training data for a Rasa model" (so it's clear its a stack model)

File: rasa/cli/interactive.py [hunk 9]
```diff
@@ -103,6 +103,7 @@ def check_training_data(args):
     story_directory, nlu_data_directory = data.get_core_nlu_directories(training_files)
     if not os.listdir(story_directory) or not os.listdir(nlu_data_directory):
         print_error(
-            "Cannot train initial Rasa model. Please provide NLU data and Core data."
+            "Cannot train initial Rasa model. Please provide NLU and Core data "
+            "using the '--data' argument."
```
Comment (erohmensing) [inline rasa/cli/interactive.py]: I guess you'll have to say "a directory of nlu and core data" right? Is there any way to pass separate files that aren't in a folder together?

File: rasa/cli/shell.py [hunk 10]
```diff
@@ -22,7 +22,7 @@ def add_subparser(
         parents=parents,
         conflict_handler="resolve",
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        help="Speak to a trained model on the command line",
+        help="Loads your trained model and lets you talk to your assistant on the command line..",
```
Comment (erohmensing) [inline rasa/cli/shell.py]:
```suggestion
        help="Loads your trained model and lets you talk to your assistant on the command line.",
```

File: rasa/cli/train.py [hunk 11]
```diff
@@ -28,15 +28,15 @@ def add_subparser(
         parents=parents,
         conflict_handler="resolve",
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        help="Train Rasa Core",
+        help="Train a Rasa Core model using your stories.",
```
Comment (erohmensing) [inline rasa/cli/train.py]:
```suggestion
        help="Trains a Rasa Core model using your stories.",
```

File: rasa/cli/train.py [hunk 12]
```diff
@@ -28,15 +28,15 @@ def add_subparser(
...
     )
     train_core_parser.set_defaults(func=train_core)
 
     train_nlu_parser = train_subparsers.add_parser(
         "nlu",
         parents=parents,
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        help="Train Rasa NLU",
+        help="Train a Rasa NLU model using your NLU data.",
```
Comment (erohmensing) [inline rasa/cli/train.py]:
```suggestion
        help="Trains a Rasa NLU model using your NLU data.",
```

File: rasa/cli/train.py [hunk 13]
```diff
@@ -15,7 +15,7 @@ def add_subparser(
 
     train_parser = subparsers.add_parser(
         "train",
-        help="Train the Rasa bot",
+        help="Train a Rasa model using your NLU data and stories.",
```
Comment (erohmensing) [inline rasa/cli/train.py]:
```suggestion
        help="Trains a Rasa model using your NLU data and stories.",
```

File: rasa/cli/x.py [hunk 14]
```diff
@@ -38,7 +38,7 @@ def add_subparser(
 
     if is_rasa_x_installed():
         # we'll only show the help msg for the command if Rasa X is actually installed
-        x_parser_args["help"] = "Start Rasa X and the Interface"
+        x_parser_args["help"] = "Starts Rasa X and the interface."
```
Comment (erohmensing) [inline rasa/cli/x.py]: Is rasa x more than the interface (at least that users would be concerned with)? maybe "starts the rasa x interface"

File: rasa/nlu/convert.py [hunk 15]
```diff
@@ -12,7 +12,10 @@ def convert_training_data(
     data_file: Text, out_file: Text, output_format: Text, language: Text
 ):
     if not os.path.exists(data_file):
-        print_error("Data file '{}' does not exist.".format(data_file))
+        print_error(
+            "Data file '{}' does not exist. Provide a valid data file using the "
```
Comment (erohmensing) [inline rasa/nlu/convert.py]:
```suggestion
            "Data file '{}' does not exist. Provide a valid NLU data file using "
```

File: rasa/nlu/convert.py [hunk 16]
```diff
@@ -12,7 +12,10 @@ def convert_training_data(
...
+            "'--data-file' argument.".format(data_file)
```
Comment (erohmensing) [inline rasa/nlu/convert.py]:
```suggestion
            "the '--data-file' argument.".format(data_file)
```

File: rasa/nlu/convert.py [hunk 17]
```diff
@@ -23,7 +26,8 @@ def convert_training_data(
         output = td.as_markdown()
     else:
         print_error(
-            "Did not recognize data format. Supported data formats: 'json' and 'md'."
+            "Did not recognize data format. Supported data formats: 'json' and 'md'. "
+            "Specify the desired data format with '--format'."
```
Comment (erohmensing) [inline rasa/nlu/convert.py]: I thought we got rid of --format in convert?

Comment (erohmensing) [reply rasa/nlu/convert.py]: if so, all this logic can go, but supported data formats should go in the "supply a valid nlu file" message

File: rasa/train.py [hunk 18]
```diff
@@ -84,7 +84,10 @@ def train(
     try:
         domain = Domain.load(domain, skill_imports)
     except InvalidDomain as e:
-        print_error(e)
+        print_error(
+            "Could not load domain due to: '{}'. To specify a valid domain path use "
+            "the '--domain' argument.".format(e)
```
Comment (erohmensing) [inline rasa/train.py]: this is just a thought and maybe doesn't go in this PR, but we should really in `InvalidDomain` check if it's a `YamlParserError` and if so lead them to YAMLint bc i've seen so many git issues from incorrectly formatted domain ymls is

File: rasa/train.py [hunk 19]
```diff
@@ -99,8 +102,8 @@ def train(
 
     if dialogue_data_not_present and nlu_data_not_present:
         print_error(
-            "No training data given. Please provide dialogue and NLU data in "
-            "order to train a Rasa model."
+            "No training data given. Please provide Core and NLU data in "
```
Comment (erohmensing) [inline rasa/train.py]:
```suggestion
            "No training data given. Please provide stories and NLU data in "
```

File: rasa/train.py [hunk 20]
```diff
@@ -190,8 +193,8 @@ def train(
         )
     else:
         print (
-            "Dialogue data / configuration did not change. "
-            "No need to retrain dialogue model."
+            "Core data / configuration did not change. "
```
Comment (erohmensing) [inline rasa/train.py]:
```suggestion
            "Stories / configuration did not change. "
```

File: rasa/train.py [hunk 21]
```diff
@@ -268,15 +271,18 @@ def train_core(
...
+            )
             return None
 
     story_directory = data.get_core_directory(stories, skill_imports)
 
     if not os.listdir(story_directory):
         print_error(
-            "No dialogue data given. Please provide dialogue data in order to "
-            "train a Rasa Core model."
+            "No Core data given. Please provide Core data in order to "
```
Comment (erohmensing) [inline rasa/train.py]:
```suggestion
            "No stories given. Please provide stories in order to "
```

File: rasa/cli/interactive.py (hunk 9 above)
Comment (tabergma) [reply rasa/cli/interactive.py]: You can actually provide multiple files like `--data data/nlu.md data/stories.md new-data/stories2.md`. So should be fine.

File: rasa/nlu/convert.py (hunk 17 above)
Comment (tabergma) [reply rasa/nlu/convert.py]: We did not. Tom pointed out that you there are not just json and markdown data formats. NLU data can also be in WIT, LUIS, or Dialogflow data format. So we kept the `--format` argument.

File: rasa/cli/interactive.py (hunk 9 above)
Comment (erohmensing) [reply rasa/cli/interactive.py]: oh, awesome!

File: rasa/train.py (hunk 18 above)
Comment (tabergma) [reply rasa/train.py]: Do you want to open an issue for that? Maybe I have time to tackle it today or tomorrow.

File: rasa/nlu/convert.py (hunk 17 above)
Comment (erohmensing) [reply rasa/nlu/convert.py]: Ah gotcha, okay. Then I would say "specify the desired output format" or something of the sort

**Pull request merged**